stl/inc/stacktrace

```diff
@@ -138,46 +138,20 @@ public:
 
     _NODISCARD __declspec(noinline) static basic_stacktrace
         current(const allocator_type& _Al = allocator_type()) noexcept {
-        _TRY_BEGIN
-        basic_stacktrace _Result{_Internal_t{}, _Max_frames, _Al};
-        const unsigned short _Actual_size = __std_stacktrace_capture(
-            1, static_cast<unsigned long>(_Max_frames), _Result._To_voidptr_array(), &_Result._Hash);
-        _Result._Frames.resize(_Actual_size);
-        return _Result;
-        _CATCH_ALL
-        return basic_stacktrace{_Al};
-        _CATCH_END
+        return _Current(1, _Max_frames, _Al);
     }
 
     _NODISCARD __declspec(noinline) static basic_stacktrace
         current(const size_type _Skip, const allocator_type& _Al = allocator_type()) noexcept {
-        _TRY_BEGIN
-        basic_stacktrace _Result{_Internal_t{}, _Max_frames, _Al};
-        const unsigned short _Actual_size = __std_stacktrace_capture(
-            _Adjust_skip(_Skip), static_cast<unsigned long>(_Max_frames), _Result._To_voidptr_array(), &_Result._Hash);
-        _Result._Frames.resize(_Actual_size);
-        return _Result;
-        _CATCH_ALL
-        return basic_stacktrace{_Al};
-        _CATCH_END
+        return _Current(_Adjust_skip(_Skip), _Max_frames, _Al);
     }
 
     _NODISCARD __declspec(noinline) static basic_stacktrace
         current(const size_type _Skip, size_type _Max_depth, const allocator_type& _Al = allocator_type()) noexcept {
-        _TRY_BEGIN
-        if (_Max_depth > _Max_frames) {
-            _Max_depth = _Max_frames;
-        }
-
-        basic_stacktrace _Result{_Internal_t{}, _Max_depth, _Al};
-
-        const unsigned short _Actual_size = __std_stacktrace_capture(
-            _Adjust_skip(_Skip), static_cast<unsigned long>(_Max_depth), _Result._To_voidptr_array(), &_Result._Hash);
-        _Result._Frames.resize(_Actual_size);
-        return _Result;
-        _CATCH_ALL
-        return basic_stacktrace{_Al};
-        _CATCH_END
+        // <NOTICE> _Current can do _Max_frames clamp inside.
+        // <NOTICE> The two _Adjust_skip in current and _Current can be merged into a single one; The efficiency gain is
+        // <NOTICE> negligible however.

```

Comment:
By merging the two _Adjust_skip:
I can make the first parameter in `_Current` naturally const; reduce a conversion from unsigned long to size_t/size_type in _Current, and `_Adjust_skip` is without doubt suitable to take `size_type` as parameter type.
Disadvantage:
In convention each helper handle the skip on its own side. This makes `_Current` an exception.